setup.cfg

```diff
@@ -107,9 +107,19 @@ ignore =
     # false positives
     B301,
     # single-letter variables
-    PDF023
+    PDF023,
     # "use 'pandas._testing' instead" in non-test code
-    PDF025
+    PDF025,
+    # If test must be a simple comparison against sys.platform or sys.version_info
+    Y002,
+    # Use "_typeshed.Self" instead of class-bound TypeVar
+    Y019,
+    # Docstrings should not be included in stubs
+    Y021,
+    # Use typing_extensions.TypeAlias for type aliases
+    Y026,
+    # Use "collections.abc.*" instead of "typing.*" (PEP 585 syntax)
+    Y027
```

Comment:
Y002: we have one `if TYPE_CHECKING`, without it mypy cannot infer some times (too many cyclic dependencies or does it hide actual errors?)
Y019: pandas does not (explicitly) depend on typeshed/typing_extension
Y021: some files have notes on how the cython and python types relate to each other
Y026: pandas does not (explicitly) depend on typing_extension
Y027: let's wait until we also do that change for the in-line annotations